src/app/pages/dry-run-preview/dry-run-preview.page.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, Input, OnInit } from '@angular/core'
+import { ModalController } from '@ionic/angular'
+import { PreapplyResponse, TezosGenericOperationError } from 'src/extension/tezos-types'
+@Component({
+  selector: 'app-dry-run-preview',
+  templateUrl: './dry-run-preview.page.html',
+  styleUrls: ['./dry-run-preview.page.scss']
+})
+export class DryRunPreviewPage implements OnInit {
+  public errors: TezosGenericOperationError[] = []
+
+  public jsonString: string | undefined
+  @Input()
+  public preapplyResponse: PreapplyResponse[] | undefined
+
+  constructor(private readonly modalController: ModalController) {}
+
+  ngOnInit(): void {
+    if (this.preapplyResponse) {
+      this.jsonString = JSON.stringify(this.preapplyResponse)
+      const operationResultErrors = this.preapplyResponse[0].contents.map(operation => {
```

Comment:
Is it ok to only take the first element of the response? Would it be possible, eg. through beacon, to have more operation groups?

Reply:
For now it's ok